test/test_gpio.c

```diff
@@ -0,0 +1,196 @@
+#include "gpio.h"
+
+#include <stdint.h>
+
+#include "unity.h"
+
+void setup_test(void) { setbuf(stdout, NULL); }
+
+void teardown_test(void) {}
+
+// gpio_init
+
+// Tests that the GPIO init works. This should simply return true 100% of time unless the test is on
+// x86 and the configuration is incorrect in which case fix it!
+void test_gpio_init(void) { TEST_ASSERT_TRUE(gpio_init()); }
+
+// gpio_init_pin
+
+// Test that a valid gpio configuration will work.
+void test_valid_gpio_init_pin(void) {
+  // Default settings for a pin.
+  GPIOSettings settings = {GPIO_DIR_IN, GPIO_STATE_LOW, GPIO_RES_NONE, GPIO_ALTFN_NONE};
+  // A pin that should be valid on all configurations.
+  GPIOAddress address = {0, 0};
+  TEST_ASSERT_TRUE(gpio_init());
+  TEST_ASSERT_TRUE(gpio_init_pin(&address, &settings));
+}
+
+// Tests a set of addresses far outside normal range.
+void test_invalid_address_gpio_init_pin(void) {
+  // Default settings for a pin.
+  GPIOSettings settings = {GPIO_DIR_IN, GPIO_STATE_LOW, GPIO_RES_NONE, GPIO_ALTFN_NONE};
+  TEST_ASSERT_TRUE(gpio_init());
+  // A port that should be invalid on all configurations.
+  GPIOAddress address = {100, 0};
```

Comment:
If our style guide does not state that we need spaces in struct initialization (`{ 100, 0 }`), we need to add that.